packages/client/lib/sync/skeleton.ts

```diff
@@ -652,6 +656,10 @@ export class Skeleton extends MetaDBManager {
         numBlocksInserted = await this.chain.putBlocks([block], true)
       } catch (e) {
         this.config.logger.error(`fillCanonicalChain putBlock error=${(e as Error).message}`)
+        if (resetHead === true && oldHead !== null) {
+          // Put original canonical head block back if reorg fails
+          await this.chain.putBlocks([oldHead], true)
```

Comment:
this should again rebuild canonical references till this oldhead right?

Reply:
I'm not 100% sure since the reorg scenario we're talking about here is just a 1 block reorg where `fillCanonical` resets the head by one block.  It resolves this hive test - `engine-transition/Transition to a Chain with Invalid Terminal Block, Higher Configured Total Difficulty` which only tests where there is a one block reorg.  That said, it seems like it should work since the reorg only resets the canonical head by 1 block I think.

Reply:
may be we extend the resetHead test i put in blockchain to restore the head and see if we can do block by number in the mid section (of reset head and the old head)

Reply:
Yep, lemme take a look and work on it some.  Your second check fixed the current broken test.